src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorCodeActionRunner.ts

```diff
@@ -6,32 +6,41 @@
 import * as vscode from 'vscode';
 import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
 import { RazorLogger } from '../RazorLogger';
-import { CodeActionResolutionRequest } from '../RPC/CodeActionResolutionRequest';
-import { CodeActionResolutionResponse } from '../RPC/CodeActionResolutionResponse';
+import { RazorCodeAction } from '../RPC/RazorCodeAction';
+import { RazorCodeActionResolutionParams } from '../RPC/RazorCodeActionResolutionParams';
 import { convertWorkspaceEditFromSerializable } from '../RPC/SerializableWorkspaceEdit';
 
 export class RazorCodeActionRunner {
+    private static readonly codeActionResolutionEndpoint = 'textDocument/codeActionResolve';
 
     constructor(
         private readonly serverClient: RazorLanguageServerClient,
         private readonly logger: RazorLogger,
     ) {}
 
     public register(): vscode.Disposable {
-        return vscode.commands.registerCommand('razor/runCodeAction', (request: CodeActionResolutionRequest) => this.runCodeAction(request), this);
```

Comment:
Oh interesting why change this?. The razor/runCodeAction command was specific to VSCode. Did we need to?

Reply:
Using just the `textDocument/codeActionResolve` for both the VS Code command + LSP endpoint ensures this is 100% just a proxy. We completely eliminate the concept of `razor/runCodeAction` to simplify the implementation.

This was my understanding from our discussion, was something else intended?

Reply:
> This was my understanding from our discussion, was something else intended?

So I was still expecting there to be a `razor/runCodeAction` in the TypeScript client, just not a `razor/resolveCodeAction` on the C# server. In your update you did both when in reality you only needed the latter. The problem with registering a general textDocument/resolveCodeAction command in VSCode is that we reserve that very common command name (no `razor` involved). In VSCode only one extension can register a command and anyone can invoke it.